### ISSUE TITLE
Errors and warnings when running npm start server mode but no errors in swagger-online-editor

### ISSUE THREAD

**Issue by pipo2152** [apigee-127/sway#136]
I am a newbie to swagger. I created my api definition. I am getting the following errors and warning for my query parameters.

```
API Errors:

  #/parameters/order_by/default: Value at index 0 not an allowable value (time, severity, occurrence): [
  #/parameters/group_by/default: Value at index 0 not an allowable value (None, uri_ref): [
  #/paths/~1all~1list_of_exceptions/get/parameters/1/default: Value at index 0 not an allowable value (time, severity, occurrence): [
  #/paths/~1all~1list_of_exceptions/get/parameters/2/default: Value at index 0 not an allowable value (None, uri_ref): [

API Warnings:

  #/parameters/top: Parameter is defined but is not used: #/parameters/top
  #/parameters/order_by: Parameter is defined but is not used: #/parameters/order_by
  #/parameters/group_by: Parameter is defined but is not used: #/parameters/group_by

4 errors and 3 warnings
```

Here is the snapshot of the parameters:

```yaml
  /all/list_of_exceptions:
    get:
      parameters:
        - $ref: '#/parameters/top'
        - $ref: '#/parameters/order_by'
        - $ref: '#/parameters/group_by'
      tags: [ALL]
      responses:
        200:
          schema:
            $ref: '#/definitions/ExceptionsList'
        default: 
          description: Error
          schema:
            $ref: '#/definitions/Error'

parameters:
  top:
    name: number_of_tips
    in: query 
    description: |
      Number of Tips to Retrieve, defaults to 'All' if not provided. However, 
      it is required if group_by is 'None'
    required: false
    type: integer
    format: int32

  order_by:
    name: sort_by
    in: query 
    required: false
    default: ['time']
    type: array
    items:
      type: string
      enum: ['time', 'severity', 'occurrence']

  group_by:
    name: group_by
    in: query 
    required: false
    default: ['None']
    type: array
    items:
      type: string
      enum: ['None', 'uri_ref']
```

**Comment by whitlockjc** [comment]
I can't see why you're getting the errors/warnings but since `swagger-editor` uses `sway`, I'm somewhat surprised.  Which editor are you speaking of?  _(I ask because SmartBear just open sourced some new tooling)_

**Comment by pipo2152** [comment]
I'm using the editor at: http://editor.swagger.io/#!/

I found that when I generated the server from the editor, it managed to change the swagger.yaml syntactically. For example, this is how my parameter looked:

  order_by:
    name: "sort_by"
    in: "query"
    required: false
    type: "array"
    items:
      type: "string"
      enum:
      - "time"
      - "severity"
      - "occurrence"
    collectionFormat: "multi"
    default: "[time]"

When I copied the contents from the swagger editor into the swagger.yaml file inside the generated node project, it works fine. I can serve the doc and view my api.
